File: src/classes/RiceCooker.ts
import {WaterLeakage, messages} from '../lib';

export class RiceCooker {
  private maxCupNumber = 10;
  private waterCupNumber = 0;
  private riceCupNumber = 0;
  private plug = false;
  private power = false;

  constructor() {}

  public addWater(cups: number) {
    WaterLeakage();
    if (this.waterCupNumber + this.riceCupNumber + cups > this.maxCupNumber) {
      throw new Error(messages['error:max-capacity']);
    }
    this.waterCupNumber += cups;
  }

  public addRice(cups: number) {
    if (this.waterCupNumber + this.riceCupNumber + cups > this.maxCupNumber) {
      throw new Error(messages['error:max-capacity']);
    }
    this.riceCupNumber += cups;
  }

  public removeWater(cups: number) {
    if (this.waterCupNumber < cups) {
      throw new Error(messages['error:not-enough-water']);
    }
    this.waterCupNumber -= cups;
  }

  public discardRice(cups: number) {
    if (this.riceCupNumber < cups) {
      throw new Error(messages['error:not-enough-rice']);
    }
    this.riceCupNumber -= cups;
  }

  public togglePlug() {
    const magicNumber = Math.round(Math.random() * 100);
    if (magicNumber > 48 && magicNumber < 52) {
      throw new Error(messages['error:plug-explosion']);
    }
    this.plug = !this.power;
  }

  public togglePower() {
    if (!this.plug) {
      throw new Error(messages['error:power-not-plugged']);
    }
    this.power = !this.power;
  }

  public getWaterCupNumber = () => this.waterCupNumber;
  public getRiceCupNumber = () => this.riceCupNumber;
  public getPlug = () => this.plug;
  public getPower = () => this.power;
}

File: riceCooker/src/main/java/org/riceCooker/classes/RiceCooker.java
package org.riceCooker.classes;

import java.util.Random;

import static org.riceCooker.utils.Messages.getMessage;
import static org.riceCooker.classes.RandomThrow.waterLeakage;

public class RiceCooker {

    private final int maxCupNumber = 10;
    private int waterCupNumber = 0;
    private int riceCupNumber = 0;
    private boolean plug = false;
    private boolean power = false;

    public RiceCooker() {}

    public void addWater(int cups) {
        waterLeakage();
        if (waterCupNumber + riceCupNumber + cups > maxCupNumber) {
            throw new RuntimeException(getMessage("error:max-capacity"));
        }
        waterCupNumber += cups;
    }

    public void addRice(int cups) {
        if (waterCupNumber + riceCupNumber + cups > maxCupNumber) {
            throw new RuntimeException(getMessage("error:max-capacity"));
        }
        riceCupNumber += cups;
    }

    public void removeWater(int cups) {
        waterLeakage();
        if (waterCupNumber < cups) {
            throw new RuntimeException(getMessage("error:not-enough-water"));
        }
        waterCupNumber -= cups;
    }

    public void discardRice(int cups) {
        if (riceCupNumber < cups) {
            throw new RuntimeException(getMessage("error:not-enough-rice"));
        }
        riceCupNumber -= cups;
    }

    public void togglePlug() {
        Random random = new Random();
        int magicNumber = random.nextInt(100);
        plug = !power;
    }

    public void togglePower() {
        if (!plug) {
            throw new RuntimeException(getMessage("error:power-not-plugged"));
        }
        power = !power;
    }

    public int getWaterCupNumber() {
        return waterCupNumber;
    }

    public int getRiceCupNumber() {
        return riceCupNumber;
    }

    public boolean getPlug() {
        return plug;
    }

    public boolean getPower() {
        return power;
    }
}

